.github/scripts/domain-check.js

```diff
@@ -0,0 +1,135 @@
+function matchesPattern(domain, filePaths) {
+    // filter files that end in .md
+    filePaths = filePaths.filter(filePath => 
+        !filePath.endsWith('.md') && 
+        !filePath.startsWith('docs/') && 
+        !filePath.startsWith('third-party-programs/')
+    );
+    // These directories contain domain specific code
+    const dirs = '(tests/unit_tests|examples|src|include/oneapi/mkl)'
+    const domains = '(blas|lapack|rng|dft)'
+    // matches changes to the domain of interest or non domain-specific code
+    const re = new RegExp(`^(${dirs}/${domain}|(?!${dirs}/${domains}))`);
+    const match = filePaths.some(filePath => re.test(filePath));
+    return match;
+}
+
+async function prFiles(github, context) {
+  const response = await github.rest.pulls.listFiles({
+    owner: context.repo.owner,
+    repo: context.repo.repo,
+    pull_number: context.payload.pull_request.number
+  });
+  const prFiles = response.data.map(file => file.filename);
+  return prFiles;
+}
+
+module.exports = async ({github, context, domain}) => {
+    if (!context.payload.pull_request) {
+        console.log('Not a pull request. Testing all domains.');
+        return true;
+    }
+    const files = await prFiles(github, context);
+    const match = matchesPattern(domain, files);
+    console.log("Domain: ", domain)
+    console.log("PR files: ", files);
+    console.log("Match: ", match);
+    return match;
+}
+
+test_patterns = [
```

Comment:
Does it make sense to add a comment indicating that this part is to test the javascript itself?

Reply:
Yes. Added comments.